File: android/src/main/java/com/wxwx/flutter_alibc/PluginUtil.java
package com.wxwx.flutter_alibc;

import static com.wxwx.flutter_alibc.PluginConstants.*;
import com.alibaba.baichuan.android.trade.model.OpenType;
import com.alibaba.baichuan.trade.biz.applink.adapter.AlibcFailModeType;
import com.alibaba.baichuan.trade.biz.core.taoke.AlibcTaokeParams;

import java.util.Map;

/**
 * @Author karedem
 * @Date 2019/9/10 11:42
 * @Description 映射返回对应值
**/
public class PluginUtil {

    public static OpenType getOpenType(String open){
        if (Auto_OpenType.equals(open)){
            return OpenType.Auto;
        }else {
           return OpenType.Native;
        }
    }

    public static String getClientType(String client){
        if (client.equals(Tmall_ClientType)){
            return "tmall";
        }else {
            return "taobao";
        }
    }

    public static AlibcFailModeType getFailModeType(String mode){
        if (JumpH5_FailMode.equals(mode)){
            return AlibcFailModeType.AlibcNativeFailModeJumpH5;
        }else if (JumpDownloadPage_FailMode.equals(mode)){
            return AlibcFailModeType.AlibcNativeFailModeJumpDOWNLOAD;
        }else {
            return AlibcFailModeType.AlibcNativeFailModeNONE;
        }
    }

    public static AlibcTaokeParams getTaokeParams(Map<String, Object> taokePar){
        String pid = (String) taokePar.get("pid");
        AlibcTaokeParams taokeParams = new AlibcTaokeParams("", "", "");
        if (pid != null){
            taokeParams.setPid(pid);
        }
        Object extParams = taokePar.get("extParams");
        //TODO 其他参数待添加
        return taokeParams;
    }

}

File: android/src/main/kotlin/com/wxwx/flutter_alibc/PluginUtil.kt
package com.wxwx.flutter_alibc

import com.alibaba.baichuan.android.trade.model.OpenType
import com.alibaba.baichuan.trade.biz.applink.adapter.AlibcFailModeType
import com.alibaba.baichuan.trade.biz.core.taoke.AlibcTaokeParams
import com.wxwx.flutter_alibc.PluginConstants.*
import com.wxwx.flutter_alibc.PluginConstants.Companion as Constants

class PluginUtil {
    companion object{
        fun getOpenType(open: String?): OpenType? {
            return if (Constants.Auto_OpenType == open) {
                OpenType.Auto
            } else {
                OpenType.Native
            }
        }

        fun getClientType(client: String): String? {
            return if (client == Constants.Tmall_ClientType) {
                "tmall"
            } else {
                "taobao"
            }
        }

        fun getFailModeType(mode: String?): AlibcFailModeType? {
            return if (Constants.JumpH5_FailMode == mode) {
                AlibcFailModeType.AlibcNativeFailModeJumpH5
            } else if (Constants.JumpDownloadPage_FailMode == mode) {
                AlibcFailModeType.AlibcNativeFailModeJumpDOWNLOAD
            } else {
                AlibcFailModeType.AlibcNativeFailModeNONE
            }
        }

        fun getTaokeParams(taokePar: Map<String?, Any?>): AlibcTaokeParams? {
            val pid = taokePar["pid"] as String?
            val taokeParams = AlibcTaokeParams("", "", "")
            if (pid != null) {
                taokeParams.setPid(pid)
            }
            val extParams = taokePar["extParams"]
            //TODO 其他参数待添加
            return taokeParams
        }
    }

}